TwitchClient/Channel.swift

```diff
@@ -1,27 +0,0 @@
-//
-//  Channel.swift
-//  TwitchClient
-//
-//  Created by Ahmed kotb on 4/8/17.
-//  Copyright © 2017 Ahmed Kotb. All rights reserved.
-//
-
-import UIKit
-
-class Channel: NSObject {
-    
-    var status : String?
-    var language : String?
-    var displayName : String?
-    var logo : String?
-    
-    func populateChannelWithData(DataDictionary : NSDictionary) -> Channel
-    {
-        self.status = DataDictionary["status"] as? String
-        self.language = DataDictionary["language"] as? String
-        self.displayName = DataDictionary["display_name"] as? String
-        self.logo = DataDictionary["logo"] as? String
-        return self
```

Comment:
Would it make sense to return an object here if all its fields are `nil`? Would it be useful? I think it's better to think which of these fields are actually optional and which are important for the model to have meaning and if they're not there then return `nil` instead of an object.

Topic to read in general: domain-driven design and benefits of immutability in Swift or in functional programming in general.